app/screens/search/search.js

```diff
@@ -81,28 +78,22 @@ class Search extends Component {
     }
 
     componentDidMount() {
-        this.refs.searchBar.focus();
-    }
-
-    shouldComponentUpdate(nextProps, nextState) {
-        return (
-            this.props.recent !== nextProps.recent ||
-            this.props.posts !== nextProps.posts ||
-            this.state !== nextState
-        );
+        if (this.refs.searchBar) {
+            this.refs.searchBar.focus();
+        }
     }
 
     componentDidUpdate(prevProps) {
         const {searchingStatus: status, recent} = this.props;
         const {searchingStatus: prevStatus} = prevProps;
-        const recentLenght = recent.length;
+        const recentLength = recent.length;
         const shouldScroll = prevStatus !== status && (status === RequestStatus.SUCCESS || status === RequestStatus.STARTED);
 
-        if (shouldScroll) {
+        if (shouldScroll && this.refs.list) {
```

Comment:
`this.refs.list` should always be set here since it's always set by render, shouldn't it?